### PR TITLE
ING-646: Handle query write statements in read-only requests.

### PR DESCRIPTION
**Pull request by brett19**

### CODE REVIEW

File: cbqueryx/queryrespreader.go [hunk 1]
```diff
@@ -122,27 +125,28 @@ func (r *queryRespReader) parseError(errJson *queryErrorJson) *ServerError {
 	if errCodeGroup == 5 {
 		err = ErrInternalServerError
 	}
-	if errCodeGroup == 12 || errCodeGroup == 14 && errCode != 12004 && errCode != 12016 {
+	if errCodeGroup == 12 || errCodeGroup == 14 {
```
Comment (Westwooo) [inline cbqueryx/queryrespreader.go:128]: Are these custom couchbase server error codes?

File: cbqueryx/queryrespreader.go [hunk 2]
```diff
@@ -122,27 +125,28 @@ func (r *queryRespReader) parseError(errJson *queryErrorJson) *ServerError {
 	if errCodeGroup == 5 {
 		err = ErrInternalServerError
 	}
-	if errCodeGroup == 12 || errCodeGroup == 14 && errCode != 12004 && errCode != 12016 {
```
Comment (Westwooo) [inline cbqueryx/queryrespreader.go:125]: Do we need to add a new if statement for  `12004` or is this alright to be `ErrIndexFailure` now?

File: cbqueryx/queryrespreader.go (hunk 1 above)
Comment (chvck) [reply cbqueryx/queryrespreader.go:128]: They're codes specific to the query service, and are included as a part of the "errors" field in the response payload.

File: cbqueryx/queryrespreader.go (hunk 2 above)
Comment (brett19) [reply cbqueryx/queryrespreader.go:125]: You're correct, although the existing code was actually already wrong, so I've filed an independent issue to address the fact that our 12004 cut-out wasn't actually working as it was likely intended to: https://couchbasecloud.atlassian.net/browse/ING-655

**Review by chvck — APPROVED** [review]

**Pull request merged**